File: benchmarks/vm/12_float_list_loop.py
import time

t0 = time.monotonic()
values = []
current = 0.0
for _ in range(5000):
    values.append(current)
    current += 1.5
for j in range(len(values)):
    values[j] = values[j] + 0.25
total = 0.0
for k in range(len(values)):
    total += values[k]
total_ms = (time.monotonic() - t0) * 1000
print(f"total = {int(total)}")
print(f"BENCH_RESULT: {total_ms:.3f} ms")

File: benchmarks/vm/12_float_list_loop.rb
t0 = Process.clock_gettime(Process::CLOCK_MONOTONIC)
values = []
current = 0.0
5_000.times do
  values << current
  current += 1.5
end
values.length.times do |j|
  values[j] = values[j] + 0.25
end
total = 0.0
values.length.times do |k|
  total += values[k]
end
total_ms = (Process.clock_gettime(Process::CLOCK_MONOTONIC) - t0) * 1000
puts "total = #{total.to_i}"
puts format("BENCH_RESULT: %.3f ms", total_ms)
